plutarch-test/plutarch-test.cabal

```diff
@@ -0,0 +1,133 @@
+cabal-version: 2.4
+name:          plutarch-test
+version:       1.1.0
+
+flag development
+  description: Enable tracing functions within plutarch.
+  manual:      True
+  default:     False
+
+common c
+  default-language:   Haskell2010
+  default-extensions:
+    NoStarIsType
+    BangPatterns
+    BinaryLiterals
+    ConstrainedClassMethods
+    ConstraintKinds
+    DataKinds
+    DeriveAnyClass
+    DeriveDataTypeable
+    DeriveFoldable
+    DeriveFunctor
+    DeriveGeneric
+    DeriveLift
+    DeriveTraversable
+    DerivingStrategies
+    DerivingVia
+    DoAndIfThenElse
+    EmptyCase
+    EmptyDataDecls
+    EmptyDataDeriving
+    ExistentialQuantification
+    ExplicitForAll
+    FlexibleContexts
+    FlexibleInstances
+    ForeignFunctionInterface
+    GADTSyntax
+    GeneralisedNewtypeDeriving
+    HexFloatLiterals
+    ImplicitPrelude
+    InstanceSigs
+    KindSignatures
+    LambdaCase
+    MonomorphismRestriction
+    MultiParamTypeClasses
+    NamedFieldPuns
+    NamedWildCards
+    NumericUnderscores
+    OverloadedStrings
+    PartialTypeSignatures
+    PatternGuards
+    PolyKinds
+    PostfixOperators
+    RankNTypes
+    RelaxedPolyRec
+    ScopedTypeVariables
+    StandaloneDeriving
+    StandaloneKindSignatures
+    TraditionalRecordSyntax
+    TupleSections
+    TypeApplications
+    TypeFamilies
+    TypeOperators
+    TypeSynonymInstances
+    ViewPatterns
+
+  ghc-options:
+    -Wall -Wcompat -Wincomplete-uni-patterns -Wno-unused-do-bind
+    -Wno-partial-type-signatures -Wmissing-export-lists -Werror
+    -Wincomplete-record-updates -Wmissing-deriving-strategies
+    -Wno-name-shadowing -Wunused-foralls -fprint-explicit-foralls
+    -fprint-explicit-kinds
+
+common deps 
+  build-depends:
+    , aeson >= 2.0
+    , base
+    , bytestring
+    , transformers
+    , sydtest
+    , sydtest-discover
+    , filepath
+    , generics-sop
+    , plutarch
+    , plutarch-benchmark
+    , plutus-core
+    , plutus-ledger-api
+    , plutus-tx
+    , rank2classes        >=1.4.4
+    , tasty
+    , tasty-golden
+    , tasty-hunit
+    , tasty-quickcheck
+    , text
+    , transformers
+
+library
+  import: c, deps
+  hs-source-dirs: src
+  exposed-modules: Plutarch.Test
+
+executable plutarch-test
+  import:             c, deps
+  type:               exitcode-stdio-1.0
+  main-is:            Main.hs
+  hs-source-dirs:     src
+
+  if impl(ghc >= 9.0)
+    other-modules:
+      Plutarch.MonadicSpec
+      Plutarch.FieldSpec
+  other-modules:
+    Spec
+    Plutarch.ApiSpec
```

Comment:
Why are you using the same prefix as `Plutarch`?

Reply:
Mostly existing convention; `Foo.hs` vs `FooSpec.hs`. Example:

- https://github.com/NorfairKing/validity/blob/master/genvalidity-time/src/Data/GenValidity/Time.hs
- https://github.com/NorfairKing/validity/blob/master/genvalidity-time/test/Data/GenValidity/TimeSpec.hs

Reply:
OK, makes sense.